example/spec/parallel/multiQueue/DiscoverGranulesMultiQueueSpec.js

```diff
@@ -0,0 +1,228 @@
+const { deleteCollection } = require('@cumulus/api-client/collections');
+const { getExecution, deleteExecution } = require('@cumulus/api-client/executions');
+const { getGranule } = require('@cumulus/api-client/granules');
+const { deleteProvider } = require('@cumulus/api-client/providers');
+const { deleteRule } = require('@cumulus/api-client/rules');
+const {
+  waitForApiStatus,
+} = require('../../helpers/apiUtils');
+const {
+  waitForGranuleAndDelete,
+} = require('../../helpers/granuleUtils');
+const {
+  createTimestampedTestId,
+  deleteFolder,
+  loadConfig,
+} = require('../../helpers/testUtils');
+
+describe('The DiscoverGranules workflow', () => {
+  let beforeAllError;
+  let bucket;
+  let collection;
+  let discoverGranulesRule;
+  let ingestGranuleExecutionArn;
+  let provider;
+  let sourcePath;
+  let discoverGranulesOutput;
+  let queueGranulesOutput;
+  let stackName;
+  let testGranuleIds;
+  let discoverGranulesExecutionArn;
+
+  beforeAll(async () => {
+    try {
+      ({ stackName, bucket } = await loadConfig());
+
+      process.env.stackName = stackName;
+      process.env.system_bucket = bucket;
+
+      const testId = createTimestampedTestId(stackName, 'DiscoverGranuleS3MultiQueue');
+
+      // The S3 path where granules will be ingested from
+      sourcePath = `${stackName}/tmp/${testId}`;
+
+      // Create the collection
+      collection = await createCollection(stackName);
+
+      // Create the provider
+      provider = await createProvider(stackName, { host: bucket });
+
+      // Upload the test data to S3
+      testGranuleIds = [`${testId}-granule1`, `${testId}-granule2`, `${testId}-granule3`];
+      await Promise.all(
+        testGranuleIds.map(async (granId) => {
+          const key = `${sourcePath}/${granId}.nc`;
+          await s3PutObject({
+            Bucket: bucket,
+            Key: key,
+            Body: 'asdf',
+          });
+        })
+      );
+
+      const ingestTime = Date.now() - 1000 * 30;
+
+      // Execute the DiscoverGranules workflow
+      discoverGranulesRule = await createOneTimeRule(
+        stackName,
+        {
+          workflow: 'DiscoverGranules',
+          collection: {
+            name: collection.name,
+            version: collection.version,
+          },
+          provider: provider.id,
+          meta: {
+            provider_path: `${sourcePath}/`,
+            queueBatchSize: 3,
+          },
+          payload: {
+            testExecutionId: testId,
+          },
+        }
+      );
+
+      discoverGranulesExecutionArn = await findExecutionArn(
+        stackName,
+        (execution) =>
+          get(execution, 'originalPayload.testExecutionId') === discoverGranulesRule.payload.testExecutionId,
+        {
+          timestamp__from: ingestTime,
+          'originalPayload.testExecutionId': discoverGranulesRule.payload.testExecutionId,
+        },
+        { timeout: 30 }
+      );
+
+      await pWaitFor(
```

Comment:
Is there any reason to use a custom helper? I believe we have a `waitForApiStatus` helper already. This is mostly a nit about code maintainability